src/base/net/portforwarder.h

```diff
@@ -48,6 +47,9 @@ namespace Net
         virtual void addPort(quint16 port) = 0;
         virtual void deletePort(quint16 port) = 0;
 
+    protected:
+        explicit PortForwarder(QObject *parent = nullptr);
```

Comment:
It is meaningless to hide constructor of abstract class since you cannot create its instance anyway.